internal/dao/patch.go

```diff
@@ -25,23 +30,31 @@ type Element struct {
 }
 
 // Build a json patch string to update PodSpec images
-func SetImageJsonPatch(images map[string]string) (string, error) {
-	elementOrders := make([]Element, 0)
-	containers := make([]Element, 0)
-	for key, value := range images {
-		elementOrders = append(elementOrders, Element{Name: key})
-		containers = append(containers, Element{Name: key, Image: value})
-	}
+func SetImageJsonPatch(spec v1.PodSpec) (string, error) {
 	jsonPatch := JsonPatch{
 		Spec: Spec{
 			Template: Template{
 				Spec: ImagesSpec{
-					SetElementOrders: elementOrders,
-					Containers:       containers,
+					SetElementOrderContainers:     extractElements(spec.Containers, false),
```

Comment:
Same issue here. We're iterating thru the containers twice. The order is already in the containers slice that tracks the name and image

Reply:
fixed : I used `var containers, initContainers` to avoid to iterate twice on elements. 

I came from Java word, I'm not used to use multi variables in return, but it's very useful.